packages/contracts/src/tx-util.ts

```diff
@@ -43,20 +45,56 @@ export class TxUtil {
     }
 
     const value = option ? (option as PayableTx).value : undefined
-    const { rawTransaction } = await web3.eth.accounts.signTransaction(
-      {
-        gasPrice,
-        gas,
+
+    // this can be replace with `web3.eth.account.signTransaction`
+    // but, there are some benefits using @ethereumjs/tx for signing tx
+    // less dependency on web3 version and more accurate gas can be use it
+    let rawTransaction
+    try {
+      const txParams = {
+        from: account.address,
         to: address,
-        value,
+        gasPrice: `0x${new BN(gasPrice).toString('hex')}`,
+        gasLimit: `0x${new BN(gas).toString('hex')}`,
+        value: `0x${new BN(value ?? 0).toString('hex')}`,
         data: tx.encodeABI(),
         nonce: option?.nonce ? +option.nonce.toString() : undefined,
-      },
-      account.privateKey,
-    )
+      }
+
+      const common = await this.getCommon(web3)
+      const unSignedTx = TransactionFactory.fromTxData(txParams, { common })
+      const signedTx = unSignedTx.sign(
+        Buffer.from(account.privateKey.substring(2), 'hex'),
+      )
+
+      if (signedTx.isSigned() && !signedTx.verifySignature()) {
+        throw new Error(`Invalid Signature`)
+      }
+
+      rawTransaction = `0x${signedTx.serialize().toString('hex')}`
+    } catch (error) {
+      logger.error(
+        `contracts/tx-utils - getting signed transaction failed : ${error}`,
+      )
+    }
+
     return rawTransaction as string
```

Comment:
I think returning string | undefined? will be safer!

Reply:
I dug deeper in `@ethereumjs/tx`. It is type safe.
Always return the string, the `serialize` method will return `0x80` at least. (= `rlp.encode(undefined)`)